compiler/src/Language/Bond/Codegen/Java/Util.hs

```diff
@@ -50,11 +50,7 @@ defaultValue :: MappingContext -> Field -> Maybe Text
 defaultValue java Field {fieldDefault = Nothing, ..} = implicitDefault fieldType
   where
     newInstance t = Just [lt|new #{getInstanceTypeName java t}()|]
-    implicitDefault (BT_Bonded t) = Just [lt|new Bonded<#{getTypeName java t}>()|]
-    -- We can't really initialize a default object for generic type
-    -- field due to the limitation of both Java and Kotlin.
-    -- I'm still looking for a solution. Please let me know if you has
-    -- any suggestions.
+    implicitDefault (BT_Bonded t) = Just [lt|new Bonded<#{getTypeName java t}>(null)|]
```

Comment:
That was intentional for the placeholder, but you have encouraged me to Do The Right Thing(tm). 

The fix for that created some weirdness that led me to a small refactor of `defaultValue`. The C# `defaultValue` returns `Maybe Text` to omit an initializer when the Bond default matches the C# implicit default, but the Java one always returns an explicit default. I dropped the `Maybe`.